### PR TITLE
Propagate message lifecycle to Smarti

### LINKED ISSUE #84: Smarti End-to-End-test
# Context
In order to ensure consistency in upgrades, we are in need of automated tests.
Though Smarti has got some unit-tests itself, integration tests are missing.

# Scenario
The "end" in end-to-end in the context of Assistify is Rocket.Chat =>
Implement tests, both for the API as well as for the UI in our RC-fork.

This story is not about comprehensive test-suite-creation, but is complete once the following succeeds

# DoD
- [ ] Via RC REST-API: Create a topic, create a corresponding request, create some messages, ensure the conversation being stored in Smarti
- [ ] Via the UI: Make sure the Smarti-Widget is being loaded once the Assistify-tab is expanded

### PR DESCRIPTION
**Pull request by mrsimpson**
fixes #84 , and https://github.com/redlink-gmbh/smarti/issues/231

This PR adds the propagates edits and deletes to Smarti in order to keep it consistent.

### CODE REVIEW

**Comment by mrsimpson** [comment]
missing: Delete of conversation

**Comment by mrsimpson** [comment]
Added delete of conversation and refactored a bit. Now, it almost looks as it should look like.

File: packages/assistify-ai/server/hooks/sendMessageToKnowledgeAdapter.js [hunk 1]
```diff
@@ -1,43 +1,76 @@
...
-		return;
+		return false;
 	}
 
-	SystemLogger.debug(`Send message ${ message._id } to knowledgeAdapter (Meteor.defer()`);
+
+}
+
+RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
+	if (isMessageRelevant(message, room)) {
```
Comment (vickyokrm) [inline packages/assistify-ai/server/hooks/sendMessageToKnowledgeAdapter.js]: This function ```isMessageRelevant``` returns **false** when certain conditions fail, should it need to explicitly return **true** if the message is relevant?

File: packages/assistify-help-request/server/hooks/sendMessageToKnowledgeAdapter.js [hunk 2]
```diff
@@ -1,53 +1,93 @@
...
+	RocketChat.settings.get('Assistify_AI_Enabled', function(key, value) {
+		knowledgeEnabled = value;
+	});
+
+	if (!knowledgeEnabled) {
+		return false;
+	}
+
+	//we only want to forward messages from requests and topics - livechats are implemented in the assistify-ai-package
+	if (!room) {
```
Comment (mrsimpson) [inline packages/assistify-help-request/server/hooks/sendMessageToKnowledgeAdapter.js]: like an optional parameter

**Comment by mrsimpson** [comment]
@vickyokrm can you please approve if this is all fine?
@ThomasRoehl this might be interesting for you to see as well though

**Pull request merged**